doc/timeline.md

```diff
@@ -0,0 +1,205 @@
+# Timeline And Keyframes
+
+The `Ra::GuiBase::Timeline` class provides display and management of `Ra::Core::Animation::KeyFramedValue`s.
+
+`Ra::Core::Animation::KeyFramedValue`s must be registered into the `Ra::GuiBase::Timeline`, which can be done using the
+`Ra::GuiBase::Timeline::registerKeyFramedValue` methods.
+
+We here consider two types of `Ra::Core::Animation::KeyFramedValue`:
+
+## Static KeyFramedValue:
+Those are `Ra::Core::Animation::KeyFramedValue`s that are an explicit part of a `Ra::Engine::Entity`, `Ra::Engine::Component` or
+`Ra::Engine::RenderObject` data, either filled upon construction or through the `Ra::GuiBase::Timeline`.
+Static `Ra::Core::Animation::KeyFramedValue`s must be registered in the `Ra::GuiBase::Timeline` after the object's
+construction.
+They usually are not bound to an *UpdateCallback* function since the object they
+belong to usually calls `Ra::Core::Animation::KeyFramedValue::at` to query the current value.
+
+Example Usage:
+```c++
+using KeyFrames = Ra::Core::Animation::KeyFramedValue<Scalar>;
+
+struct MyComponentWithKeyFrame : public Ra::Engine::Component
+{
+   MyComponentWithKeyFrame( const std::string& name, Ra::Engine::Entity* entity, Scalar value = 0_ra )
+       : Ra::Engine::Component( name, entity )
+       , m_keyframes( value, 0_ra )
+       , m_currentValue( value )
+   {
+       // use the linear interpolator instead of the default one
+       m_keyframes.setInterpolator( Ra::Core::Animation::interpolate<Scalar> );
+   }
+
+   void initialize() override {}
+   void goTo( Scalar t ) {
+       // Here we use the value at time t.
+       m_currentValue = m_keyframes.at( t );
```

Comment:
Why does `m_keyframes` couldn't store the current value of `t`, so there is no need to store/duplicate `m_currentValue`.
That would also make the code more compact on the client side.

Reply:
The thing here is that the KeyFrames are just set for some times t0, t1...tn, which are interpolated when querying the value for any time t.
In this case, m_currentValue is the interpolation result, which needs to be stored since computation may be expensive; and there is no duplication other than the one for storing it, though the compiler may be smart enough to create it in place.